libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteArrow.java

```diff
@@ -110,8 +110,10 @@
   void addUpcomingManeuverArrow(RouteProgress routeProgress) {
     boolean invalidUpcomingStepPoints = routeProgress.upcomingStepPoints() == null
             || routeProgress.upcomingStepPoints().size() < TWO_POINTS;
-    boolean invalidCurrentStepPoints =
-      routeProgress.currentLegProgress().currentStepProgress().stepPoints().size() < TWO_POINTS;
+    boolean invalidCurrentStepPoints = routeProgress.currentLegProgress() != null
+            && routeProgress.currentLegProgress().currentStepProgress() != null
+            && routeProgress.currentLegProgress().currentStepProgress().stepPoints() != null
+            && routeProgress.currentLegProgress().currentStepProgress().stepPoints().size() < TWO_POINTS;
```

Comment:
Sorry, I might've phrased it poorly - isn't below the correct check?

```suggestion
    boolean invalidCurrentStepPoints = routeProgress.currentLegProgress() == null
            || routeProgress.currentLegProgress().currentStepProgress() == null
            || routeProgress.currentLegProgress().currentStepProgress().stepPoints() == null
            || routeProgress.currentLegProgress().currentStepProgress().stepPoints().size() < TWO_POINTS;
```

Reply:
The original logic indicated invalidCurrentStepPoints was true if the size was less than TWO_POINTS. I suppose it could be argued that a null on the way to getting that value is in effect zero points and therefore invalidCurrentStepPoints would be true. At the time I was thinking about avoiding the NPE. I'll make the adjustment you suggest.